x-pack/plugins/observability/server/domain/services/date_range.ts

```diff
@@ -0,0 +1,72 @@
+/*
```

Comment:
💬 Main change: toDateRange domain service that compute the DateRange from a given slo time window and (optional) provided date